Controllers/frontend/NostoTagging.php

```diff
@@ -330,4 +331,28 @@ private function fireRequest(NostoOAuthToken $token)
         }
         return $result;
     }
+
+    /**
+     * Restores the cart based on the given hash
+     *
+     * @throws Enlight_Event_Exception
+     * @throws Enlight_Exception
+     * @throws Zend_Db_Adapter_Exception
+     * @throws Exception
+     */
+    public function cartAction()
+    {
+        $hash = $this->Request()->getParam(CartRestore::CART_RESTORE_URL_PARAMETER);
+        $sessionId = Shopware()->Session()->get('sessionId');
+        if ($hash === null || $sessionId === null) {
+            return;
+        }
+        $cartRestore = new CartRestore();
```

Comment:
Before restoring the cart check if the visitor has carted items in current session. If items are found redirect user to the checkout page without restoring the "old " cart